parlai/agents/tfidf_retriever/tfidf_doc_ranker.py

```diff
@@ -17,32 +17,52 @@
 
 from . import utils
 from . import tokenizers
+from parlai.core.agents import Agent
 
 logger = logging.getLogger(__name__)
 
 
-class TfidfDocRanker(object):
+class TfidfDocRankerAgent(Agent):
```

Comment:
An agent has to implement observe/act. Probably you just want to share the object directly in the user.